code/Calendar.php

```diff
@@ -276,19 +276,31 @@ public static function getWeekdayName($value, $width = 'wide', $locale = '', $st
     {
         $result = '';
         if ((!empty($value)) || ($value === 0) || ($value === '0')) {
-            $weekday = null;
-            if (is_int($value)) {
-                $weekday = $value;
-            } elseif (is_float($value)) {
-                $weekday = intval($value);
-            } elseif (is_string($value)) {
-                if (is_numeric($value)) {
+            switch (gettype($value)) {
```

Comment:
I wasn't aware of this. Are you sure??? It's a huge breaking change for soooo many projects!